core/controllers/admin_test.py

```diff
@@ -161,3 +164,84 @@ def test_invalid_username_in_view_and_update_role(self):
+        self.editor_id = self.get_user_id_from_email(self.EDITOR_EMAIL)
+        self.exploration = self.save_new_valid_exploration(
+            self.EXP_ID, self.editor_id, end_state_name='End')
+
+        stats_services.record_answer(
+            self.EXP_ID, self.exploration.version,
+            self.exploration.init_state_name, 'TextInput',
+            stats_domain.SubmittedAnswer(
+                'first answer', 'TextInput', 0,
+                0, exp_domain.EXPLICIT_CLASSIFICATION, {},
+                'a_session_id_val', 1.0))
+
+        stats_services.record_answer(
+            self.EXP_ID, self.exploration.version,
+            self.exploration.init_state_name, 'TextInput',
+            stats_domain.SubmittedAnswer(
+                'second answer', 'TextInput', 0,
+                0, exp_domain.EXPLICIT_CLASSIFICATION, {},
+                'a_session_id_val', 1.0))
+
+    def test_data_extraction_handler(self):
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+
+        # Test that it returns all answers when 'num_answers' is 0.
+        payload = {
+            'exp_id': self.EXP_ID,
+            'exp_version': self.exploration.version,
+            'state_name': self.exploration.init_state_name,
+            'num_answers': 0
+        }
+
+        response = self.testapp.get(feconf.ADMIN_URL)
+        csrf_token = self.get_csrf_token_from_response(response)
+        response = self.post_json(
+            '/explorationdataextractionhandler', payload, csrf_token=csrf_token)
+        extracted_answers = response['data']
+        self.assertEqual(len(extracted_answers), 2)
+        self.assertEqual(extracted_answers[0]['answer'], 'first answer')
+        self.assertEqual(extracted_answers[1]['answer'], 'second answer')
+
+        # Make sure that it returns only 'num_answers' number of answers.
+        payload = {
+            'exp_id': self.EXP_ID,
+            'exp_version': self.exploration.version,
+            'state_name': self.exploration.init_state_name,
+            'num_answers': 1
+        }
+
+        response = self.testapp.get(feconf.ADMIN_URL)
+        csrf_token = self.get_csrf_token_from_response(response)
+        response = self.post_json(
+            '/explorationdataextractionhandler', payload, csrf_token=csrf_token)
+        extracted_answers = response['data']
+        self.assertEqual(len(extracted_answers), 1)
+        self.assertEqual(extracted_answers[0]['answer'], 'first answer')
+
+    def test_that_handler_raises_exception(self):
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+        payload = {
+            'exp_id': self.EXP_ID,
+            'exp_version': self.exploration.version,
+            'state_name': 'state name',
+            'num_answers': 0
+        }
+        response = self.testapp.get(feconf.ADMIN_URL)
+        csrf_token = self.get_csrf_token_from_response(response)
+        response = self.post_json(
```

Comment:
I meant checking the text of the error message that actually gets sent to the frontend. (I'm not actually sure, from a cursory reading of this test, the reason why it raises an exception.)